pallets/messages/src/lib.rs

```diff
@@ -212,7 +271,7 @@ impl<T: Config> Pallet<T> {
 	pub fn add_message(
 		provider_key: T::AccountId,
 		message_source_id: MessageSourceId,
-		payload: Vec<u8>,
+		payload: Payload,
```

Comment:
Looking good.

Question: do we need to pass in an enum here? I don't think we are doing anything with knowing if the payload is offchain or onchain.

Reply:
We serialize the payload into the message differently based on what it is. Either way, I think the type of the `payload` param would have to be something other than a vector. Thoughts?